core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -170,30 +170,28 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
    * @return a boolean indicating whether the size of the memory map and the underneath file is changed or not.
    */
   def resize(newSize: Int): Boolean = {
-    inLock(lock) {
-      val roundedNewSize = roundDownToExactMultiple(newSize, entrySize)
+    val roundedNewSize = roundDownToExactMultiple(newSize, entrySize)
```

Comment:
This isn't safe, right? You want to compute it inside the lock. It's also fast, so no point in trying to optimize it as far as I can see.

Reply:
`roundDownToExactMultiple` does not touch any shared mutable variables, there is no need to execute it with a lock held. Right? However I could follower your suggestion if you insist :)